plugins/scaffolder-backend-module-confluence-to-markdowwn/src/actions/confluence/confluenceToMarkdown.ts

```diff
@@ -0,0 +1,177 @@
+
+      // This grabs the files from Github
+      const repoFileDir = `${dirPath}/${parsedRepoUrl.filepath}`;
+      await fetchPlainAction.handler({
+        ...ctx,
+        input: {
+          url: `https://${parsedRepoUrl.resource}/${parsedRepoUrl.owner}/${parsedRepoUrl.name}`,
+          targetPath: dirPath,
+        },
+      });
+
+      for (const url of confluenceUrls) {
+        const { spacekey, title, titleWithSpaces } =
+          await createConfluenceVariables(url);
+        // This calls confluence to get the page html and page id
+        const getConfluenceDoc = await fetchConfluence(
+          `/rest/api/content?title=${title}&spaceKey=${spacekey}&expand=body.export_view`,
+          config,
+        );
+        if (getConfluenceDoc.results.length === 0) {
+          throw new InputError(
+            `Could not find document ${url}. Please check your input.`,
+          );
+        }
+        // This gets attachements for the confluence page if they exist
+        const getDocAttachments = await fetchConfluence(
+          `/rest/api/content/${getConfluenceDoc.results[0].id}/child/attachment`,
+          config,
+        );
+
+        if (getDocAttachments.results.length) {
+          fs.mkdirSync(`${dirPath}/${filePathToMkdocs}docs/img`, {
+            recursive: true,
+          });
+          productArray = await getAndWriteAttachments(
+            getDocAttachments,
+            dirPath,
+            config,
+            filePathToMkdocs,
+          );
+        }
+
+        ctx.logger.info(
+          `starting action for converting ${titleWithSpaces} from Confluence To Markdown`,
+        );
+
+        // This reads mkdocs.yml file
+        const mkdocsFileContent = await readFileAsString(repoFileDir);
+        const mkdocsFile = await YAML.parse(mkdocsFileContent);
+        ctx.logger.info(
+          `Adding new file - ${titleWithSpaces} to the current mkdocs.yml file`,
+        );
+
+        // This modifies the mkdocs.yml file
+        if (mkdocsFile !== undefined && mkdocsFile.hasOwnProperty('nav')) {
+          const { nav } = mkdocsFile;
+          if (!nav.some((i: Obj) => i.hasOwnProperty(titleWithSpaces))) {
+            nav.push({
+              [titleWithSpaces]: `${titleWithSpaces.replace(/\s+/g, '-')}.md`,
+            });
+            mkdocsFile.nav = nav;
+          } else {
+            throw new ConflictError(
+              'This document looks to exist inside the GitHub repo. Will end the action.',
+            );
+          }
+        }
+
+        await fs.writeFile(repoFileDir, YAML.stringify(mkdocsFile));
+
+        // This grabs the confluence html and converts it to markdown and adds attachments
+        const html = getConfluenceDoc.results[0].body.export_view.value;
+        const markdownToPublish = NodeHtmlMarkdown.translate(html);
+        let newString: string = markdownToPublish;
+        productArray.forEach((product: string[]) => {
+          const regex = product[1].includes('pdf')
```

Comment:
This check if `product[1]` include pdf but the regexes below operate on `product[0]`, is that intentional?

Reply:
Both product[0] and product[1] would have `.pdf` in the name. I  have updated it to product[0] for consistency